tests/refarm_pagination/test_views.py

```diff
@@ -0,0 +1,65 @@
+from contextlib import contextmanager
+from unittest import mock
+
+from django.conf import settings
+from django.test import TestCase
+from refarm_pagination import views, pagination
+
+from tests.catalog.models import MockProduct
+
+
+class PaginationContext(TestCase):
+
+    fixtures = ['catalog.json']
+
+    def context(self, **kwargs):
+        return views.PaginationContext(**{
+            'url': '',
+            'number': 1,
+            'per_page': 1,
+            'objects': MockProduct.objects.all(),
+            **kwargs
+        }).context()
+
+    @contextmanager
+    def mock_neighbor_pairs(self, url, number):
+        with \
+            mock.patch('refarm_pagination.pagination.NeighborPage') as mocked_page,\
+            mock.patch('refarm_pagination.views.NeighborPages') as mocked_pages:
+
+            mocked_page.url = lambda _: url
+            mocked_page.number = number
+
+            get_neighbors = lambda: [mocked_page] * (settings.PAGINATION_NEIGHBORS // 2)
+            pages = mocked_pages.return_value
+            pages.next_neighbors = pages.prev_neighbors = get_neighbors
+
+            yield mocked_page, mocked_pages
+
+    def test_page_number(self):
+        with mock.patch('django.core.paginator.Page') as mocked:
+            number = mocked.return_value.number = 1
```

Comment:
i'm not good in mocking objects. So, explain plz one detail for me.
Here we patch page object with this code: `mocked.return_value.number`
But in method ahead we patch it with this code: `mocked_pages.return_value`.

Why code is different?

Reply:
@duker33 , `mocked.return_value` is a value, that will be returned when we call/instantiate `mocked` object
```python
In [34]: M
Out[34]: unittest.mock.Mock

In [35]: m = M()

In [36]: m
Out[36]: <Mock id='4459582240'>

In [38]: m()
Out[38]: <Mock name='mock()' id='4458025032'>

In [39]: m() == m.return_value
Out[39]: True
```